src/query/functions/src/scalars/array.rs

```diff
@@ -149,6 +149,30 @@ pub fn register(registry: &mut FunctionRegistry) {
         ),
     );
 
+    registry.register_2_arg_core::<ArrayType<NullableType<GenericType<0>>>, GenericType<0>, UInt64Type, _, _>(
+        "indexof",
+        FunctionProperty::default(),
+        |_, _| FunctionDomain::Full,
+        vectorize_with_builder_2_arg::<ArrayType<NullableType<GenericType<0>>>, GenericType<0>, UInt64Type>(
+            |arr, val, output, _| {
+                let len = arr.len();
+                let mut matched = false;
+                for idx in 0..len {
+                    if let Some(Some(item)) = arr.index(idx) {
+                        if item == val {
```

Comment:
How should we handle the case where val is `NULL`?

Reply:
Directly return 0

Reply:
0 is not a valid index, how about return `NULL`?

Reply:
Now is this:
```
select indexof([1,2,null,2,null], null)
----
3
```
I think that's better. :+1: